src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ComposeStore.targets

```diff
@@ -259,40 +271,23 @@ Copyright (c) .NET Foundation. All rights reserved.
     <MakeDir Directories="$(PublishDir)" />
     <MakeDir  Directories="$(StoreStagingDir)"/>
 
-    <!-- Create a Project to be used for restores-->
-    <PropertyGroup>
-      <RestoreProjContent>
-        <![CDATA[
-<Project Sdk="Microsoft.NET.Sdk">
-</Project>
-]]>
-      </RestoreProjContent>
-    </PropertyGroup>
-
-    <WriteLinesToFile
-        File="$(RestoreProj)"
-        Lines="$(RestoreProjContent)"
-        Overwrite="true" />
   </Target>
 
-  <Target Name="PrepforRestoreForComposeStore">
+  <Target Name="PrepforRestoreForComposeStore"
+          DependsOnTargets="_DefaultMicrosoftNETPlatformLibrary">
 
     <PropertyGroup>
       <StorePackageVersionForFolderName>$(StorePackageVersion.Replace('*','-'))</StorePackageVersionForFolderName>
       <StoreWorkerWorkingDir>$([System.IO.Path]::Combine($(ComposeWorkingDir),"$(StorePackageName)_$(StorePackageVersionForFolderName)"))</StoreWorkerWorkingDir>
-      <_PackageProjFile>$([System.IO.Path]::Combine($(StoreWorkerWorkingDir), "Package.csproj"))</_PackageProjFile>
-      <ProjectAssetsFile>$(StoreWorkerWorkingDir)\project.assets.json</ProjectAssetsFile>
-      <_PackageProjContent>
-        <![CDATA[
-<Project Sdk="Microsoft.NET.Sdk">
-  <ItemGroup>
-    <PackageReference Include="$(StorePackageName)" Version="$(StorePackageVersion)"/>
-  </ItemGroup>
-</Project>    
-]]>
-      </_PackageProjContent>
+      <_PackageProjFile>$([System.IO.Path]::Combine($(StoreWorkerWorkingDir), "Restore.csproj"))</_PackageProjFile>
+      <BaseIntermediateOutputPath>$(StoreWorkerWorkingDir)\</BaseIntermediateOutputPath>
+      <ProjectAssetsFile>$(BaseIntermediateOutputPath)\project.assets.json</ProjectAssetsFile>
     </PropertyGroup>
     
+    <PropertyGroup>
+      <PackagesToPrune>$(MicrosoftNETPlatformLibrary)</PackagesToPrune>
+      <SelfContained Condition="'$(SelfContained)' == ''">true</SelfContained>
```

Comment:
because as of today it only gets set for self contained executable projects:
https://github.com/dotnet/sdk/blob/master/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.RuntimeIdentifierInference.targets#L107